### ISSUE TITLE
Very strange behaviour of OS.has_touchscreen_hint()

### ISSUE THREAD

**Issue by dploeger** [godotengine/godot#49113]
**Godot version:**
3.3

**OS/device including version:**
Android-Emulator running a Pixel with Android 30

**Issue description:**
This is a hard one, I guess, anyways...

We're developing a game using EgoVenture and it is using `OS.has_touchscreen_hint()` to determine wether the game's running on a touchscreen or not.

Now, in this function it hides inventory buttons depending on that: https://github.com/deep-entertainment/egoventure/blob/master/addons/egoventure/nodes/inventory/inventory.gd#L30

The example games works perfect, but in a new project we're developing, it doesn't work. If I run it using an Android emulator and set a breakpoint there, `is_touch` is set to false:

<img width="1047" alt="Bildschirmfoto 2021-05-26 um 19 44 57" src="https://user-images.githubusercontent.com/203453/119707806-d9c80d80-be5b-11eb-9ed1-07bb45297dd9.png">

If I simply add another (actually bogus) code line there, it magically becomes true:

<img width="1043" alt="Bildschirmfoto 2021-05-26 um 19 46 26" src="https://user-images.githubusercontent.com/203453/119707852-e9dfed00-be5b-11eb-9bfd-12d81afb177b.png">

I tried creating a sample project for this issue, but can't reproduce it there. I'd say it's a race condition but according to the code, the Android platform should always return true for OS.has_touchscreen_hint().

So basically, I'm stumped. Any ideas?

**Steps to reproduce:**

If I'd only knew! :(

**Minimal reproduction project:**

Sadly nothing. :(

**Comment by Calinou** [comment]
@m4gr3d Any ideas what could be causing this?

**Comment by dploeger** [comment]
Thanks for the suggestion, but that behaves similarly strange. When used in _ready, `OS.has_feature("mobile")` returns true. When used in another part of the code, which is also code by `_ready` from another script, it returns false.

**Comment by Calinou** [comment]
> Thanks for the suggestion, but that behaves similarly strange. When used in _ready, `OS.has_feature("mobile")` returns true. When used in another part of the code, which is also code by `_ready` from another script, it returns false.

What about `OS.get_name() == "Android" || OS.get_name == "iPhone"`?

**Comment by dploeger** [comment]
> Does `var is_mobile = OS.get_name() == "Android" || OS.get_name() == "iPhone"` exhibit the same issue?

Using that works. Are iPads also returning "iPhone" for OS.get_name()? If so it would be a valid workaround.

**Comment by Calinou** [comment]
> Using that works. Are iPads also returning "iPhone" for OS.get_name()? If so it would be a valid workaround.

Yes, "iPhone" is the OS name returned by the OSIPhone class.

**Comment by dploeger** [comment]
Okay, I will implement that into EgoVenture for the time being.

**Comment by dploeger** [comment]
Thanks a lot.

**Comment by m4gr3d** [comment]
> @m4gr3d Any ideas what could be causing this?

@Calinou @dploeger That's a strange issue indeed. Both Android and iOS [return `true` for this method](https://github.com/godotengine/godot/blob/e9909b763af21adf0bfffca94280aa4c7ac7a601/platform/android/os_android.cpp#L660) so it should always be true.

@dploeger Based on your sample code, I'm wondering if this could be a gdscript initialization issue.
cc @vnen

**Comment by dploeger** [comment]
Well, at least on iPadOS 15.5, OS.get_name is returning "iOS" instead of "iPhone" on Godot 3.3
